src/mixins/ui/lasso-tool-ui.js

```diff
@@ -19,6 +19,92 @@ const MathExt = Draw.Math
 const DragEpsilon = 0.1 // this is a screen-space epsilon to validate that drag events are doing something
 // this can be relatively big since we're talking about screen coords
 
+/**
+ * Enum class that defines the different contexts in which a shape handler's destroy() method
+ * might be called. The enum context can be used in ShapeHandler-derived detroy() method to
+ * destroy the appropriate shapes that may have been created by the handler. For example,
+ * temporary shapes may be created while creating a new shape (i.e. a simple circle for each
+ * vertex for visual emphasis). The shape handler may want to only delete the temporary shapes
+ * in its destroy() method and not its primary shape.
+ */
+class DestroyType {
```

Comment:
Ended up going through the process of including a DestroyType enum class to identify the context in which `ShapeHandler::destroy()` methods are called. This is useful to include a reason for a `LASSO_TOOL_DRAW_ENDED` signal.